pkg/server/status/recorder.go

```diff
@@ -336,34 +336,49 @@ func (mr *MetricsRecorder) MarshalJSON() ([]byte, error) {
 // ScrapeIntoPrometheus updates the passed-in prometheusExporter's metrics
 // snapshot.
 func (mr *MetricsRecorder) ScrapeIntoPrometheus(pm *metric.PrometheusExporter) {
-	mr.mu.RLock()
-	defer mr.mu.RUnlock()
-	if mr.mu.nodeRegistry == nil {
-		// We haven't yet processed initialization information; output nothing.
-		if log.V(1) {
-			log.Warning(context.TODO(), "MetricsRecorder asked to scrape metrics before NodeID allocation")
+	mr.ScrapeIntoPrometheusWithStaticLabels(false)(pm)
+}
+
+func (mr *MetricsRecorder) ScrapeIntoPrometheusWithStaticLabels(
+	useStaticLabels bool,
+) func(pm *metric.PrometheusExporter) {
+	return func(pm *metric.PrometheusExporter) {
+		includeChildMetrics := ChildMetricsEnabled.Get(&mr.settings.SV)
+		includeAggregateMetrics := includeAggregateMetricsEnabled.Get(&mr.settings.SV)
+		scrapeOptions := []metric.ScrapeOption{
+			metric.WithIncludeChildMetrics(includeChildMetrics),
+			metric.WithIncludeAggregateMetrics(includeAggregateMetrics),
+			metric.WithUseStaticLabels(useStaticLabels),
+		}
+		mr.mu.RLock()
```

Comment:
I'm not sure if this makes any difference, but I like when objects are locked as the first stage of a function, even if the critical section is further along. It's almost as if it's part of the function signature which I find useful.

Reply:
Done.